tests/semantic_analyser.cpp

```diff
@@ -908,25 +890,34 @@ TEST(semantic_analyser, call_delete)
   test("kprobe:f { @y[5, 4] = 5; delete(@y, ((uint8)5, (uint64)4)); }");
 
   test_error("kprobe:f { delete(1); }", R"(
-stdin:1:12-20: ERROR: delete() expects a map for the first argument and a key for the second argument e.g. `delete(@my_map, 1);`
+stdin:1:12-21: ERROR: delete() requires 2 arguments (1 provided)
 kprobe:f { delete(1); }
+           ~~~~~~~~~
+)");
+
+  test_error("kprobe:f { delete(1, 1); }", R"(
+stdin:1:12-20: ERROR: delete() expects a map argument
```

Comment:
can we include the argument index in the error message? e.g. `expects a map as the first argument` or `as argument number 1`

Reply:
I can make this annotation on the actual first argument.

Reply:
Done